pkg/controller/registry/resolver/resolver.go

```diff
@@ -129,7 +140,57 @@ func (r *OperatorsV1alpha1Resolver) ResolveSteps(namespace string, sourceQuerier
 		}
 	}
 
-	return steps, updatedSubs, nil
+	// process other operators first
+	bundleLookups := []*v1alpha1.BundleLookup{}
+
+	// prune operators that have in progress bundle image jobs
+	for bundleImageInfo := range gen.PendingOperators() {
+		// TODO: this is not ideal...
+		ips, err := r.listInstallPlans(namespace)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		for _, ip := range ips {
+			for _, lookup := range ip.Status.BundleLookups {
```

Comment:
I think I'm missing why this is needed - can't you just return the whole set of pending operators as `BundleLookups` that the installplan should unpack?

Reply:
I do have a TODO here to investigate improving. Without it install plans get duplicated due to the next loop doing a resolve (again) and repeating. The only way that I can see to avoid this is by checking all the install plans for the image.